### ISSUE TITLE
macOS: Mod Loader Freezes With Script-Based Mods Active (1.1.2)

### ISSUE THREAD

**Issue by SupakugaiGames** [Rubberduckycooly/Sonic-CD-11-Decompilation#118]
Before I begin, I want to say thanks to everyone involved with the project for presenting what I believe is the definitive version of Sonic CD. I love this project so much, and having a native Mac version is a huge blessing.

With that said, upon building the newest version on macOS (testing on 11.4 on an M1 Air with and without Rosetta), I noticed that the mod loader would freeze upon loading certain mods. If starting up the game with those mods active, it would show a black screen. The game was still operational as I was able to go back to the dev menu and disable mods. I theorized that this is related to custom scripts because [Green Hill Zone,](https://gamebanana.com/mods/149690) a script-less mod, worked perfectly fine. I also made a basic mod where the only change was one edited script, and it froze like mentioned earlier.

When trying to force a level to load while either the [Super Sonic in Sonic CD](https://gamebanana.com/mods/293589) or [Rolling Sound Effect](https://gamebanana.com/mods/298201) mods are active, it resulted in this:

![image](https://user-images.githubusercontent.com/77683570/123626690-4fe4d900-d83b-11eb-8bdd-09a108f337a4.png)
A blue sky with uncontrollable characters.

What is odd about this to me is that RSDKv4's mod loader works perfectly fine under the same build/test conditions. I started comparing code between the two projects to see if I could figure out what was wrong, but I wanted to bring this here as an issue in case someone with more experience working on this project could provide a better solution than I could come up with (I'm not very experienced with C++).

Long story short: RSDKv3 1.1.2's mod loader refuses to load scripts on macOS, RSDKv4 works perfectly fine.

**Comment by MegAmi24** [comment]
[Install the decompiled scripts](https://github.com/Rubberduckycooly/Sonic-CD-2011-Script-Decompilation/archive/refs/heads/main.zip)
